Src/CSharpier/ConfigurationFileOptions.cs

```diff
@@ -17,58 +17,68 @@ public class ConfigurationFileOptions
         public bool UseTabs { get; init; }
         public EndOfLine EndOfLine { get; init; }
 
-        public static ConfigurationFileOptions Create(string rootPath, IFileSystem fileSystem)
-        {
-            ConfigurationFileOptions ReadJson(string path)
-            {
-                return JsonConvert.DeserializeObject<ConfigurationFileOptions>(
-                    fileSystem.File.ReadAllText(path)
-                );
-            }
-
-            ConfigurationFileOptions ReadYaml(string path)
+        public static ConfigurationFileOptions Create(
+            string baseDirectoryPath,
+            IFileSystem fileSystem
+        ) {
+            while (true)
             {
-                var deserializer = new DeserializerBuilder().WithNamingConvention(
-                        CamelCaseNamingConvention.Instance
-                    )
-                    .Build();
+                var potentialPath = fileSystem.Path.Combine(baseDirectoryPath, ".csharpierrc");
 
-                return deserializer.Deserialize<ConfigurationFileOptions>(
-                    fileSystem.File.ReadAllText(path)
-                );
-            }
+                if (fileSystem.File.Exists(potentialPath))
+                {
+                    var contents = fileSystem.File.ReadAllText(potentialPath);
+                    if (contents.TrimStart().StartsWith("{"))
+                    {
+                        return ReadJson(potentialPath, fileSystem);
+                    }
 
-            var potentialPath = fileSystem.Path.Combine(rootPath, ".csharpierrc");
+                    return ReadYaml(potentialPath, fileSystem);
+                }
 
-            if (fileSystem.File.Exists(potentialPath))
-            {
-                var contents = fileSystem.File.ReadAllText(potentialPath);
-                if (contents.TrimStart().StartsWith("{"))
+                var jsonExtensionPath = potentialPath + ".json";
+                if (fileSystem.File.Exists(jsonExtensionPath))
                 {
-                    return ReadJson(potentialPath);
+                    return ReadJson(jsonExtensionPath, fileSystem);
                 }
 
-                return ReadYaml(potentialPath);
-            }
+                var yamlExtensionPaths = new[] { potentialPath + ".yaml", potentialPath + ".yml" };
```

Comment:
Then you can use that `directoryInfo` like so:
```cs
directoryInfo
  .EnumerateFiles(".csharpierrc*", SearchOption.TopDirectoryOnly)
  .Select(file => /* do pattern matching here with the suffix */)
```
to avoid multiple calls to disk.